### PR TITLE
Implement Element.getAttributeNames

### PR DESCRIPTION
**Pull request by nox**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9064)

### CODE REVIEW

File: components/script/dom/element.rs [hunk 1]
```diff
@@ -1201,6 +1201,11 @@ impl ElementMethods for Element {
         self.attr_list.or_init(|| NamedNodeMap::new(&window_from_node(self), self))
     }
 
+    // https://dom.spec.whatwg.org/#interface-element
+    fn GetAttributeNames(&self) -> Vec<DOMString> {
```
Comment (Manishearth) [inline components/script/dom/element.rs]: https://dom.spec.whatwg.org/#dom-element-getattributenames

**Comment by Manishearth** [comment]
r=me with the updated spec link

**Comment by eefriedman** [comment]
```
  │ FAIL [expected PASS] Element interface: element must inherit property "setAttributeNode" with the proper type (20)
  │   → assert_inherits: property "setAttributeNode" not found in prototype chain
```

etc.  Looks like the expected results just need to be updated.

**Comment by nox** [comment]
@eefriedman Yes. I'm looking into idlharness.js to not include the index in the test name (there is no need for it, overloaded operations share only one property).

**Comment by nox** [comment]
Blocked by https://github.com/w3c/testharness.js/pull/170.

**Comment by nox** [comment]
I changed the indices in the expectation file instead of staying blocked on my testharness.js PR.

@bors-servo r=Manishearth

**Comment by KiChjang** [comment]
```
Ran 4606 tests finished in 229.0 seconds.
  • 4604 ran as expected. 1374 tests skipped.
  • 2 tests had unexpected subtest results

Tests with unexpected results:
  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  │ FAIL [expected PASS] Element interface: document.createElement("noscript") must inherit property "removeAttributeNode" with the proper type (22)
  │   → assert_inherits: property "removeAttributeNode" not found in prototype chain
  │ FAIL [expected PASS] Element interface: document.createElement("noscript") must inherit property "query" with the proper type (34)
  │   → assert_inherits: property "query" not found in prototype chain
  │ FAIL [expected PASS] Element interface: document.createElement("noscript") must inherit property "queryAll" with the proper type (35)
  │   → assert_inherits: property "queryAll" not found in prototype chain
  │ 
  │ IdlInterface.prototype.test_interface_of/<@http://web-platform.test:8000/resources/idlharness.js:1467:25
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1381:20
  │ test@http://web-platform.test:8000/resources/testharness.js:495:9
  │ IdlInterface.prototype.test_interface_of@http://web-platform.test:8000/resources/idlharness.js:1461:1
  │ IdlInterface.prototype.test_object@http://web-platform.test:8000/resources/idlharness.js:1373:9
  │ IdlArray.prototype.test/<@http://web-platform.test:8000/resources/idlharness.js:353:17
  │ IdlArray.prototype.test@http://web-platform.test:8000/resources/idlharness.js:351:13
  └ window.onload@http://web-platform.test:8000/html/dom/interfaces.html:237:3

  ▶ Unexpected subtest result in /websockets/cookies/005.html:
  └ PASS [expected FAIL] WebSockets: setting HttpOnly cookies in ws response, checking ws request
```

**Comment by jdm** [comment]
There's no reason the websocket test should be passing with this change :( Probably file that as a new intermittent pass.

**Comment by nox** [comment]
@bors-servo r=Manishearth

I changed the expectations and filed #9772.

**Pull request merged**